tests/gradients/finite_diff/test_spsa_gradient.py

```diff
@@ -85,9 +85,7 @@ def test_same_seeds(self):
         second_direction = _rademacher_sampler(ids, num, rng=rng)
         assert np.allclose(first_direction, second_direction)
 
-    @pytest.mark.parametrize(
-        "ids, num", [(list(range(5)), 5), ([0, 2, 4], 5), ([0], 1), ([2, 3], 5)]
-    )
+    @pytest.mark.parametrize("ids, num", [([0, 2, 4], 5), ([0], 1), ([2, 3], 5)])
```

Comment:
Instead of splitting this into two tests that do the exact same thing, you can do the following:

```suggestion
    @pytest.mark.parametrize("ids, num", [pytest.param(list(range(5)), 5, marks=pytest.mark.xfail(reason="Failure after updating rng salt to 0.41.0 [sc-90962]")), ([0, 2, 4], 5), ([0], 1), ([2, 3], 5)])
```